services/reproducible-central/reproducible-central.service.js

```diff
@@ -0,0 +1,91 @@
+import Joi from 'joi'
+import { BaseJsonService, pathParams } from '../index.js'
+
+const schema = Joi.object({
+  color: Joi.string().required(),
+  message: Joi.string().required(),
+}).required()
+
+export default class ReproducibleCentral extends BaseJsonService {
+  static category = 'dependencies'
+
+  static route = {
+    base: 'reproducible-central/a',
+    pattern: ':groupId/:artifactId/:version',
+  }
+
+  static openApi = {
+    '/reproducible-central/a/{groupId}/{artifactId}/{version}': {
+      get: {
+        summary: 'Reproducible Central',
+        parameters: pathParams(
+          {
+            name: 'groupId',
+            example: 'org.apache.maven',
+          },
+          {
+            name: 'artifactId',
+            example: 'maven-core',
+          },
+          {
+            name: 'version',
+            example: '3.9.9',
+          },
+        ),
+      },
+    },
+  }
+
+  static defaultBadgeData = {
+    label: 'Reproducible Builds',
+    labelColor: '1e5b96',
+  }
+
+  static render({ message, color }) {
+    return {
+      message,
+      color,
+    }
+  }
+
+  async fetch({ groupId, artifactId, version }) {
+    return this._requestJson({
+      schema,
+      url: `https://jvm-repo-rebuild.github.io/reproducible-central/badge/artifact/${groupId.replace(/\./g, '/')}/${artifactId}/${version}.json`,
+    })
+  }
+
+  async fetchIndex({ groupId, artifactId, version }) {
+    const { res } = await this._requestFetcher({
+      url: `https://jvm-repo-rebuild.github.io/reproducible-central/badge/artifact/${groupId.replace(/\./g, '/')}/${artifactId}/index.html`,
+    })
+    if (res.statusCode === 404) {
+      return { message: 'unknown ga', color: 'orange' } // unknown project ga, whatever version
+    } else {
+      return { message: version, color: 'grey' } // unknown version for a known project ga
+    }
+  }
+
+  async handle({ groupId, artifactId, version }) {
+    if (version.endsWith('-SNAPSHOT')) {
+      return this.constructor.render({ message: 'SNAPSHOT', color: 'grey' })
+    }
```

Comment:
Again, what are we trying to tell the user in this case? If I'm expecting a reproducibility score but getting back `SNAPSHOT` seems odd to me.

Reply:
Reproducible Builds checking for SNAPSHTO can't have been checked yet, by definition: trying to get an adapted feedback

Reply:
Again, I think I'd say we want a short error that explains that. Maybe "not analysed" or "snapshot builds are not analysed" or something like that. Essentially: An explanation of why can we not return a result for this, as opposed to a description of what the user submitted. Some other similar examples from elsewhere:

https://github.com/badges/shields/blob/2a20f813df8281f1fc00be6fa7ea1563bee4adcb/services/crates/crates-downloads.service.js#L86-L91
https://github.com/badges/shields/blob/2a20f813df8281f1fc00be6fa7ea1563bee4adcb/services/uptimerobot/uptimerobot-base.js#L43-L48
https://github.com/badges/shields/blob/2a20f813df8281f1fc00be6fa7ea1563bee4adcb/services/gem/gem-downloads.service.js#L129-L133

Reply:
ok, changed the message